src/io/flutter/editor/NativeEditorNotificationProvider.java

```diff
@@ -42,29 +41,38 @@ public EditorNotificationPanel createNotificationPanel(@NotNull VirtualFile file
     if (!file.isInLocalFileSystem()) {
       return null;
     }
-    final VirtualFile root = findRootDir(file);
+    return createPanelForFile(file, findRootDir(file, myProject.getBaseDir()));
+  }
+
+  private EditorNotificationPanel createPanelForFile(VirtualFile file, VirtualFile root) {
     if (root == null) {
       return null;
     }
+    return createPanelForAction(file, root, getActionName(root));
+  }
+
+  private EditorNotificationPanel createPanelForAction(VirtualFile file, VirtualFile root, String actionName) {
+    if (actionName == null) {
+      return null;
+    }
+    NativeEditorActionsPanel panel = new NativeEditorActionsPanel(file, root, actionName);
+    return panel.isValidForFile() ? panel : null;
+  }
 
-    final String actionName;
+  private static String getActionName(VirtualFile root) {
```

Comment:
can `root` be null? If no, consider a `@NotNull` annotation

Reply:
Good catch. Thanks.